xls/dslx/stdlib/std.x

```diff
@@ -199,35 +199,69 @@ fn smul_test() {
     assert_eq(s4:0b1111, smul(s2:0b11, s2:0b01));
 }
 
-// Calculate x / y one bit at a time. This is an alternative to using
-// the division operator '/' which may not synthesize nicely.
-pub fn iterative_div<N: u32, DN: u32 = {N * u32:2}>(x: uN[N], y: uN[N]) -> uN[N] {
-    let init_shift_amount = ((N as uN[N]) - uN[N]:1);
-    let x = x as uN[DN];
-
-    let (_, _, _, div_result) = for
-        (idx, (shifted_y, shifted_index_bit, running_product, running_result)) in range(u32:0, N) {
-        // Increment running_result by current power of 2
-        // if the prodcut running_result * y < x.
-        let inc_running_result = running_result | shifted_index_bit;
-        let inc_running_product = running_product + shifted_y;
-        let (running_result, running_product) = if inc_running_product <= x {
-            (inc_running_result, inc_running_product)
+// Returns unsigned division of `n` (N bits) and `d` (M bits) as quotient (N bits) and remainder (M
+// bits).
+// If dividing by `0`, returns alls `1`s for quotient and `n` for remainder.
+// Based on the implementation in //xls/ir/abstract_evaluator:UDivMod;
```

Comment:
Rather than citing that implementation, can we point out that this is actually binary long division?